openwisp_monitoring/device/api/views.py

```diff
@@ -200,6 +201,20 @@ def _write(self, request, pk):
                 metric.write(client['mac'])
             if created:
                 self._create_clients_graph(metric)
+        if 'resources' in data:
+            if 'load' in data['resources']:
+                self.resources_metric_write(
+                    pk, ct, key='load', value=100 * data['resources']['load'][1]
+                )
+            if 'disk_usage' in data['resources']:
+                usage = 0
+                for disk in data['resources']['disk_usage']:
+                    usage += disk['percent_used']
+                self.resources_metric_write(pk, ct, key='disk_usage', value=usage)
+            if 'memory' in data['resources']:
+                value = data['resources']['memory']
+                used = 1 - (value['free'] - value['buffered']) / value['total']
+                self.resources_metric_write(pk, ct, key='memory', value=100 * used)
```

Comment:
Better do the full calculus in one line and then just `self.resources_metric_write(pk, ct, key='memory', value=used)`

Reply:
@nemesisdesign Same here, what value do you think we should save?

Reply:
@PabloCastellano, I need to save 3 metrics with different keys and values what I think would be best to avoid redundancy is store the keys and values in a dictionary and then use a for metric_write. Let me know, if this seems fine.